packages/service-client/odsp-client/README.md

```diff
@@ -0,0 +1,101 @@
+## @fluid-experimental/odsp-client
+
+The odsp-client package provides a simple and powerful way to consume collaborative Fluid data with the ODSP service.
+
+## Using odsp-client
+
+The odsp-client package has a OdspClient class that allows you to interact with Fluid
+
+```typescript
+import { OdspClient } from "@fluid-experimental/odsp-client";
+```
+
+### Example usage
+
+```typescript
+import { OdspClient, OdspConnectionConfig, OdspClientProps } from "@fluid-experimental/odsp-client";
+
+const connectionConfig: OdspConnectionConfig = {
+	tokenProvider: "<YOUR_TOKEN_PROVIDER>",
+	siteUrl: "<SITE_URL>",
+	driveId: "<RAAS_DRIVE_ID>",
+	path: "<FOLDER_PATH>",
+};
+
+export const clientProps: OdspClientProps = {
+	connection: connectionConfig,
+};
+
+const client = new OdspClient(clientProps);
+```
+
+### Experimental Features
+
+`OdspClient` supports the ability to instantiate with experimental features enabled.
+These features are experimental in nature and should **NOT** be used in production applications.
+To learn more, see [Experimental Features](https://fluidframework.com/docs/build/experimental-features/).
+
+## Fluid Containers
+
+A Container instance is a organizational unit within Fluid. Each Container instance has a connection to the defined Fluid Service and contains a collection of collaborative objects.
+
+Containers are created and identified by unique IDs. Management and storage of these IDs are the responsibility of the developer.
+
+## Defining Fluid Containers
+
+Fluid Containers are defined by a schema. The schema includes initial properties of the Container as well as what collaborative objects can be dynamically created.
+
+```typescript
+const containerSchema = {
+	initialObjects: {
+		/* ... */
+	},
+	dynamicObjectTypes: [
+		/*...*/
+	],
+};
+const OdspClient = new OdspClient(clientProps);
+const { container, services } = await OdspClient.createContainer(containerSchema);
```

Comment:
So there is no way to specify file name, right?
And it's possible (for load path) to specify itemID that does not match URL & Path info in clientProps.
Both seem wrong to me.

While we might want not to expose filename concept (and just use guide), users are interacting with storage through GraphAPI as well, and might want to name their files. This might come handy for various workflows like search, eDiscovery that clients might need to deal with. I think it should be possible to specify a file name.

Reply:
todo: will iterate on this in upcoming PRs. Possibly moving site url, drive id, file path, fileName close to container attachment.